src/Application/Actions/Donations/Create.php

```diff
@@ -68,6 +68,27 @@ protected function action(): Response
 
         try {
             $donation = $this->donationRepository->buildFromApiRequest($donationData);
+
+            // Optionally set these donation fields if available
+            if (isset($donationData->billingPostalAddress)) {
```

Comment:
We'll need to revisit whether we actually need to collect billing and postal address separately for Gift Aid + card processing purposes. This doesn't align with the WIP for the frontend but may still be the best option.